openlibrary/catalog/add_book/load_book.py

```diff
@@ -92,26 +152,39 @@ def walk_redirects(obj, seen):
             seen.add(obj['key'])
         return obj
 
-    q = {'type': '/type/author', 'name': name}  # FIXME should have no limit
-    reply = list(web.ctx.site.things(q))
+    # Try for an 'exact' (case-insensitive) name match, but fall back to alternate_names,
+    # then last name with identical birth and death dates (that are not themselves `None`).
+    queries = [
+        {"type": "/type/author", "name~": author["name"]},
+        {"type": "/type/author", "alternate_names~": author["name"]},
+        {
+            "type": "/type/author",
+            "name~": f"* {author['name'].split()[-1]}",
+            "birth_date": author.get("birth_date", -1),
+            "death_date": author.get("death_date", -1),
```

Comment:
If it's okay to change `openlibrary.core.helpers.extract_year()` to support 3 digit dates, we could probably use something like:
```
>>> pattern = re.compile(r'(?<!\d)(\d{3,4})(?!\d)')
>>> pattern.search("999-01-01").group()
'999'
>>> pattern.search("9999-01-01").group()
'9999'
>>> pattern.search("September 1st, 408").group()
'408'
>>> # This will choke and need to be handled, but it's to show it won't match three to four digit subsets of larger numbers.
>>> pattern.search("September 1st, 40008").group()
Traceback (most recent call last):
  File "<input>", line 1, in <module>
    pattern.search("September 1st, 40008").group()
     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'group'
```
(Though obviously not accessing the `.group()` method without checking that there's a result first.)

Or it would be easy to add another function.

Reply:
We talked ; this would unfortunately cause issues with the thing query, since `*342*` would match too much on the backend! Eg authors born in `1342`.